cmd/wardend/cmd/gen-keychains.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/spf13/cobra"
+
+	address "cosmossdk.io/core/address"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/server"
+	"github.com/cosmos/cosmos-sdk/x/genutil"
+	genutiltypes "github.com/cosmos/cosmos-sdk/x/genutil/types"
+
+	wardentypes "github.com/warden-protocol/wardenprotocol/warden/x/warden/types/v1beta2"
+)
+
+func AddGenesisKeychainCmd(defaultNodeHome string, addressCodec address.Codec) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add-genesis-keychain <creator> <description>",
+		Short: "Add a Keychain to the genesis file",
+		Args:  cobra.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			config.SetRoot(clientCtx.HomeDir)
+
+			return addGenesisKeychain(
+				clientCtx.Codec,
+				config.GenesisFile(),
+				args[0],
+				args[1],
+			)
+		},
+	}
+
+	cmd.Flags().String(flags.FlagHome, defaultNodeHome, "The application home directory")
+	cmd.Flags().String(flags.FlagKeyringBackend, flags.DefaultKeyringBackend, "Select keyring's backend (os|file|kwallet|pass|test)")
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+func addGenesisKeychain(
+	cdc codec.Codec,
+	genesisFileURL string,
+	creator string,
+	description string,
+) error {
+	appState, appGenesis, err := genutiltypes.GenesisStateFromGenFile(genesisFileURL)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal genesis state: %w", err)
```

Comment:
should we use wrapped errors to know more precisely where they are coming from?

Reply:
I actually refactored a bit to reduce the amount of duplicated code, and I wrapped all the errors

consider that you run singularly these commands such as in `wardend genesis add-genesis space xxx` so the error surface is not that big